### DISCUSSION TITLE
Can we make writeData public instead of protected?

### DISCUSSION THREAD

**Discussion by T-vK** [pschatzmann/arduino-midi#10] [Q&A]
I'm currently trying to write a class (extending from MidiCommon) that encapsulated the functionality of BLE Midi, Apple Midi and serial Midi.

The easiest way (in terms of sending) would have been to do sth. like this:

```
void MultiMidi::writeData(MidiMessage *msg, int len){
    serialStreamOut->writeData(msg, len);
    bleServer->writeData(msg, len);
    appleServer->writeData(msg, len);
}
```

But unfortunately `AppleMidiServer::writeData` and `MidiStreamOut` have `writeData` set up as a protected method meaning I can't call them as shown above.
For `MidiBleServer::writeData`, surprisingly it is not protected. But still, for the other classes it creates great inconveniences for my use case.

It would be nice if we could simply make that method public consistently. Sort of like the internal `Print` class has `write` set up publicly.

I would be willing to create a PR for this.

**Comment by pschatzmann** [comment]
Fine with me: How about adding a new public write() method to MidiCommon which calls writeData

**Reply by T-vK** [reply]
Sure, that would work for me too!
